src/encoded/static/components/search.js

```diff
@@ -277,39 +277,64 @@ globals.listingViews.register(Biosample, 'Biosample');
 
 
 /**
- * Renders both Experiment and FunctionalCharacterizationExperiment search results.
+ * Renders both Experiment, FunctionalCharacterizationExperiment, TransgenicEnhancerExperiment search results.
  */
 const ExperimentComponent = (props, reactContext) => {
     const { context: result, cartControls, mode } = props;
     let synchronizations;
 
-    // Determine whether object is Experiment or FunctionalCharacterizationExperiment.
+    // Determine whether object is Experiment, FunctionalCharacterizationExperiment, or TransgenicEnhancerExperiment.
     const experimentType = result['@type'][0];
     const isFunctionalExperiment = experimentType === 'FunctionalCharacterizationExperiment';
-    const displayType = isFunctionalExperiment ? 'Functional Characterization Experiment' : 'Experiment';
+    const isEnhancerExperiment = experimentType === 'TransgenicEnhancerExperiment';
+    let displayType;
+    if (isFunctionalExperiment) {
+        displayType = 'Functional Characterization Experiment';
+    } else if (isEnhancerExperiment) {
+        displayType = 'Transgenic Enhancer Experiment';
+    } else {
+        displayType = 'Experiment';
+    }
 
     // Collect all biosamples associated with the experiment. This array can contain duplicate
     // biosamples, but no null entries.
     let biosamples = [];
     const treatments = [];
 
-    if (result.replicates && result.replicates.length > 0) {
-        biosamples = _.compact(result.replicates.map(replicate => replicate.library && replicate.library.biosample));
-        // flatten treatment array of arrays
-        _.compact(biosamples.map(biosample => biosample.treatments)).forEach(treatment => treatment.forEach(t => treatments.push(t)));
+    if (isEnhancerExperiment) {
+        if (result.biosamples && result.biosamples.length > 0) {
+            biosamples = _.compact(result.biosamples.map(sample => biosamples));
```

Comment:
We need to deduplicate with `_.compact` below because `result.replicates.biosample` can really have duplicates. But I don‘t think `result.biosamples` can have duplicates, so you can just assign `result.biosamples` unmodified to `biosamples`.